### ISSUE TITLE
Not finding correct libpython on mingw-w64

### ISSUE THREAD

**Issue by ktbarrett** [ktbarrett/find_libpython#1]
```
$ find_libpython
C:/msys64/mingw64/lib/python3.8/config-3.8/libpython3.8.dll.a
```

Correct location:
```
C:/msys64/mingw64/bin/libpython3.8.dll
```

**Comment by ktbarrett** [comment]
<details><summary>candidate names</summary>

```
$ find_libpython --candidate-names
libpython3.8.dll.a
libpython3.8.dll
python3.8.dll
python3.dll
python.dll
```

</details>

<details><summary>candidate paths</summary>

```
$ find_libpython --candidate-paths
c:/msys64/mingw64/lib/python3.8/config-3.8/libpython3.8.dll.a
c:/msys64/mingw64/lib/python3.8/config-3.8/libpython3.8.dll
c:/msys64/mingw64/lib/python3.8/config-3.8/python3.8.dll
c:/msys64/mingw64/lib/python3.8/config-3.8/python3.dll
c:/msys64/mingw64/lib/python3.8/config-3.8/python.dll
C:/msys64/mingw64/lib/python3.8/config-3.8/libpython3.8.dll.a
C:/msys64/mingw64/lib/python3.8/config-3.8/libpython3.8.dll
C:/msys64/mingw64/lib/python3.8/config-3.8/python3.8.dll
C:/msys64/mingw64/lib/python3.8/config-3.8/python3.dll
C:/msys64/mingw64/lib/python3.8/config-3.8/python.dll
c:/msys64/mingw64/lib/libpython3.8.dll.a
c:/msys64/mingw64/lib/libpython3.8.dll
c:/msys64/mingw64/lib/python3.8.dll
c:/msys64/mingw64/lib/python3.dll
c:/msys64/mingw64/lib/python.dll
c:/msys64/mingw64/bin/libpython3.8.dll.a
c:/msys64/mingw64/bin/libpython3.8.dll
c:/msys64/mingw64/bin/python3.8.dll
c:/msys64/mingw64/bin/python3.dll
c:/msys64/mingw64/bin/python.dll
c:/msys64/mingw64/libpython3.8.dll.a
c:/msys64/mingw64/libpython3.8.dll
c:/msys64/mingw64/python3.8.dll
c:/msys64/mingw64/python3.dll
c:/msys64/mingw64/python.dll
C:\msys64\mingw64\bin/libpython3.8.dll
```

</details>

The last path in the candidate paths is the correct one. The one returned by these lines.
https://github.com/ktbarrett/find_libpython/blob/2602a7c1be61c4b01a0376d388b47b4a8c8ca721/src/find_libpython/__init__.py#L235-L237

This should be moved up. If the library can be found immediately, there is no reason to continue searching.

Additionally, candidate names contains `libpython38.dll.a`. AFAIK this is *not* a viable dynamic library name and should not be considered. `get_config_var("LDLIBRARY")` and `get_config_var("LIBRARY")` return libraries that are archives due to the python in question *not* being compiled with `--enable-shared`.

**Comment by ktbarrett** [comment]
Reverted, so reopening.

**Comment by ktbarrett** [comment]
Fixing #7 will break this. The issue is that `LDLIBRARY` is the library to use during linking. In msys2, cygwin, and apparently mingw32/64 this means the import library ([see here](https://github.com/python/cpython/blob/206cbdab16cb054e859597a562e2f6ab35e99766/configure#L5766-L5773)). On msys2, mingw64, and cygwin we shoudl be using the `DLLLIBRARY` variable to get the path to the dynamic library. We currently aren't separately testing for mingw64 in the source.

**Comment by ktbarrett** [comment]
Closed by #16.
